Failed to create an XPathFactory during getMetadata

I'd like to report a runtime error that I'm seeing in a microservice after trying to migrate to Quarkus 3.0.3 and Tika 2.0.0.

I have been able to recreate the issue using the same file but a new Quarkus project created by the Quarkus CLI. I don't use any custom configurations or business logic. All I need to do is create a project that uses the `quarkus-tika` and `quarkus-container-image-jib` extensions which tries to grab the metadata of files like so:

```
parser.getMetadata(inputStream, "application/pdf")
```

When running the project on my host machine (either through the test runner or dev mode) the code runs as expected and I'm able to scrape the metadata of whatever documents I can throw at it. However if I take the same code, build it into a container image using JIB, and run that I then encounter an error for a specific input file. I'd like to share the exact file that causes the issue, but it is generated by a third-party and may have test environment data I'd rather not risk sharing. I hope the stack trace is enough to begin troubleshooting, if not let me know and I'll see if I can find any public documents that cause the same crash.

I was hoping to include a fully reproducible project as a part of this issue but since it seems contingent on a potentially sensitive input file this was the best I could think of to start this discussion. Please let me know if I'm missing any important clues and I'll try to help out. 

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: java.lang.ExceptionInInitializerError
        at org.apache.pdfbox.filter.FilterFactory.<init>(FilterFactory.java:42)
        at org.apache.pdfbox.filter.FilterFactory.<clinit>(FilterFactory.java:35)
        at org.apache.pdfbox.cos.COSStream.getFilterList(COSStream.java:304)
        at org.apache.pdfbox.cos.COSStream.createInputStream(COSStream.java:175)
        at org.apache.pdfbox.cos.COSStream.createInputStream(COSStream.java:163)
        at org.apache.pdfbox.pdfparser.PDFXrefStreamParser.<init>(PDFXrefStreamParser.java:56)
        at org.apache.pdfbox.pdfparser.COSParser.parseXrefStream(COSParser.java:2865)
        at org.apache.pdfbox.pdfparser.COSParser.parseXrefObjStream(COSParser.java:435)
        at org.apache.pdfbox.pdfparser.COSParser.parseXref(COSParser.java:386)
        at org.apache.pdfbox.pdfparser.COSParser.retrieveTrailer(COSParser.java:255)
        at org.apache.pdfbox.pdfparser.PDFParser.initialParse(PDFParser.java:173)
        at org.apache.pdfbox.pdfparser.PDFParser.parse(PDFParser.java:226)
        at org.apache.pdfbox.pdmodel.PDDocument.load(PDDocument.java:1232)
        at org.apache.pdfbox.pdmodel.PDDocument.load(PDDocument.java:1206)
        at org.apache.tika.parser.pdf.PDFParser.getPDDocument(PDFParser.java:317)
        at org.apache.tika.parser.pdf.PDFParser.parse(PDFParser.java:172)
        at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:298)
        at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:298)
        at org.apache.tika.parser.AutoDetectParser.parse(AutoDetectParser.java:195)
        at io.quarkus.tika.TikaParser.parseStream(TikaParser.java:85)
        at io.quarkus.tika.TikaParser.getMetadata(TikaParser.java:68)
        at org.acme.GreetingCommand.run(GreetingCommand.java:47)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:132)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: XPathFactory#newInstance() failed to create an XPathFactory for the default object model: http://java.sun.com/jaxp/xpath/dom with the XPathFactoryConfigurationException: javax.xml.xpath.XPathFactoryConfigurationException: No XPathFctory implementation found for the object model: http://java.sun.com/jaxp/xpath/dom
        at javax.xml.xpath.XPathFactory.newInstance(Unknown Source)
        at org.apache.pdfbox.filter.DCTFilter.<clinit>(DCTFilter.java:63)
        ... 42 more
```

--- a/runtime/pom.xml
+++ b/runtime/pom.xml
@@ -94,6 +94,12 @@
         <dependency>
             <groupId>org.apache.tika</groupId>
             <artifactId>tika-parser-xml-module</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>xerces</groupId>
+                    <artifactId>xercesImpl</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.apache.tika</groupId>

